js/canvas.js

```diff
@@ -0,0 +1,367 @@
+import { arrayOfRectangles } from './rectangles.js';
+import { canvasWindowIndent, startedRectangleFillColor, redColorInHex, stickyDistance } from './shared/constants.js';
+
+export class Canvas {
+  constructor(selectedCanvas) {
+    this.canvas = selectedCanvas;
+    this.ctx = this.canvas.getContext('2d');
+    this.draggingRect;
+    this.isDragOk = false;
+    this.offsetX;
+    this.offsetY;
+  }
+
+  start() {
+    this.setSize();
+    this.setOffsets_X_and_Y();
+    this.drawRectangles();
+    this.canvas.addEventListener('mousedown', this.mouseDownHandler.bind(this));
+    this.canvas.addEventListener('mousemove', this.mouseMoveHandler.bind(this));
+    this.canvas.addEventListener('mouseup', this.mouseUpHandler.bind(this));
+  }
+
+  setSize() {
+    const { innerWidth: width, innerHeight: height } = window;
+
+    this.canvas.width = width - canvasWindowIndent;
+    this.canvas.height = height - canvasWindowIndent;
+  }
+
+  setOffsets_X_and_Y() {
+    const domRect = canvas.getBoundingClientRect();
+    const { left, top } = domRect;
+
+    this.offsetX = left;
+    this.offsetY = top;
+  }
+
+  drawRectangles() {
+    this.clear();
+
+    arrayOfRectangles.forEach(rect => {
+      if (rect.isSticked && !rect.isCrossed) {
+        const { stickedX, stickedY, width, height } = rect;
+
+        this.ctx.fillStyle = rect.fillColor;
+        this.ctx.fillRect(stickedX, stickedY, width, height);
+      } else {
+        const { x, y, width, height } = rect;
+
+        this.ctx.fillStyle = rect.fillColor;
+        this.ctx.fillRect(x, y, width, height);
+      }
```

Comment:
lines 43-46 looks like duplicate for lines 48-51.

`
 arrayOfRectangles.forEach(rect => {
    const { x, y, stickedX, stickedY, width, height } = rect;
    this.ctx.fillStyle = rect.fillColor;
    const coorX = rect.isSticked && !rect.isCrossed ? stickedX : x;
    const coorY = rect.isSticked && !rect.isCrossed ? stickedY : y;
    this.ctx.fillRect(coorX, coorY, width, height);
}
`